dxr/app.py

```diff
@@ -170,6 +170,20 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
     try:
         count_and_results = query.results(offset, limit)
+        # If we're asked to redirect and there's a single result, redirect to the result.
+        if (request.values.get('redirect') == 'true' and
+            count_and_results['result_count'] == 1):
+            _, path, line, _ = next(count_and_results['results'])
+            line = line[0][0] if line else None
```

Comment:
It's worth considering whether it's desirable to redirect the user to a found FILE-domain result, e.g. when they're doing a `path:` search just to find out whether any such files exist. In those cases, they quite possibly aren't concerned with the contents of the file at all, and the slowness of loading a large file might be worse than what we have now. But I can likely be persuaded on this point. What do you think?